hutch_python/user_load.py

```diff
@@ -50,3 +56,181 @@ def get_user_objs(load, *, ask_on_failure=True):
                 )
 
     return objs
+
+
+def update_objs(
+    obj_ns: utils.HelpfulNamespace,
+    entry: str,
+    attrs: dict,
+    fn: Callable[[BaseInterface, Union[dict, list[str]]], None]
+) -> None:
+    """
+    Helper function for updating object namespaces
+    obj_ns is a HelpfulNamespace, which may include other
+    HelpfulNamespace's.  As a result this must be semi-recursive.
+
+    Parameters
+    ----------
+    obj_ns : HelpfulNamespace
+        Namespace of all objects
+
+    entry : str
+        Name of device to edit
+
+    attrs : list[str]
+        List of attributes to add to whitelist
+
+    fn : Callable[[BaseInterface, dict], None]
+        A function that applies a change to a singular device
+    """
+    dev = getattr(obj_ns, entry, None)
+    if dev:  # look for explicit device
+        fn(dev, attrs)
+
+        return
+
+    # look for devices with class
+    for dev in obj_ns:
+        if isinstance(dev, utils.HelpfulNamespace):
+            update_objs(dev, entry, attrs, fn)
```

Comment:
How do we get to this branch in a useful way? Do the nested namespaces have to have some collusion-level coordination on the entry string? I feel like my brain rolled off the code at some point

Reply:
If I'm understanding the question right: the way we hit this recursive step is because I pass in the `all_objects` namespace, which also has HelpfulNamespaces inside it (`sim`, `m`, `x`, etc).  

Reading over your comment again I'm very not sure what the question is.

Reply:
I'm trying to remember what I was thinking about last week, I think I may have just misunderstood the code